### ISSUE TITLE
Presence of `- ` in `alt` attribute results in extraneous, invalid `<p>`

### ISSUE THREAD

**Issue by coreyward** [probablyup/markdown-to-jsx#382]
The following markdown results in the content of the `figure` element being wrapped in a `<p>` tag when it should not:

Input:
```md
Test

<figure>![Uh - oh ](//placehold.it/200x200)</figure>

Test
```

Output:
```html
<div>
  <p>Test</p>
  <figure>
    <p> <!-- 👈 👀 -->
      <img alt="Uh - oh " src="//placehold.it/200x200">
    </p>
  </figure>
  <p>Test</p>
</div>
```

**Comment by Prabhakar-Poudel** [comment]
overwrite your `p` tag like so

```jsx
export const YourParagraph = ({ children, ...props }) =>  {
  if (hildren[0]?.type?.name === 'figure') return <>{children}</>
  return <p {...props}>{children}</p>
}
```

**Comment by coreyward** [comment]
@Prabhakar-Poudel Yep, there are some strategies like this for working around the issue, but it's still a bug.

**Comment by nilsmich** [comment]
Hi,
Thank you Prabhakar-Poudel for the fix. I agree with coreyward, it would be nice to have a fix for this. 

I had the same problem with other tags. E.g. 
`
<ul>
    <li>some - text</li>
</ul>
`
is rendered as  
`
<ul>
    <li>**<p>**some - text**</p>**</li>
</ul>
`. 

I used a more general fix as I mainly use the package for HTML parsing. 

`
export const ParagraphFix = ({children, ...props}) => {
  if (children[0]?.type?.name === 'p') return <p {...props}>{children}</p>
  return <>{children}</>
}
export const overrides: MarkdownToJSX.Overrides = {
  p: Paragraph,
  ...
}
`

May help some of you ;)

**Comment by meglio** [comment]
Same problem with even a simpler scenario:

```html
<p>Some text - </p>
```

**Comment by quantizor** [comment]
Looks to be fixed in v9
